app/controllers/catalog_controller.rb

```diff
@@ -456,31 +463,31 @@ def active_sort_fields
     config.add_sort_field("date_modified_desc") do |field|
       field.label = "date modified \u25BC"
       field.sort = "date_modified_dtsi desc"
-      field.if = ->(controller, field) { controller.current_user }
+      field.if = :can_see_extra_sort_orders?
```

Comment:
Again I would cosider skipping the cover method, and just doing:

    field.if = ->(controller, field) { controller.can?(:see_admin_pages) }

But... I see how you're using it in multiple places, maybe it makes sense to add the extra indirection here, I dunno, up to you. But note that the above should work fine. I prob would have done that. Just like `controller.current_user` was repeated previously, without a cover method, which was fine.

Reply:
No, you're 100% right.